### ISSUE TITLE
nvcc fails due to illegal options

### ISSUE THREAD

**Issue by gkoundry** [SJTU-IPADS/PowerInfer#23]
When I try to build this I get the following error:
`nvcc fatal   : Unknown option 'Wmissing-declarations'`

The cmake files seem to be set up to pass gcc options to nvcc which AFAIK are not supported.

**Comment by nick-lambdalabs** [comment]
+1 getting the same here

```
[  7%] Building CXX object CMakeFiles/llama.dir/llama.cpp.o
nvc++-Error-Unknown switch: -Wmissing-declarations
nvc++-Error-Unknown switch: -Wmissing-noreturn
gmake[2]: *** [CMakeFiles/llama.dir/build.make:76: CMakeFiles/llama.dir/llama.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:676: CMakeFiles/llama.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
``` 

I am trying to build inside an NVidia HPC docker container:

```
sudo docker run --gpus all -it nvcr.io/nvidia/nvhpc:23.11-devel-cuda_multi-ubuntu22.04
```

**Comment by hodlen** [comment]
We have found the root cause of these errors that CMake passes host compiling arguments to `nvcc` and fixed it via #161. Please feel free to reopen this issue if the bug persists.
